frontend/src/framework/internal/components/RightSettingsPanel/private-components/RealizationFilterSettings/realizationFilterSettings.tsx

```diff
@@ -215,10 +276,17 @@ export const RealizationFilterSettings: React.FC<RealizationFilterSettingsProps>
             selections.includeOrExcludeFilter !== realizationFilter.getIncludeOrExcludeFilter();
 
         // Update the unsaved changes state
-        setEnsembleIdentStringHasUnsavedChangesMap({
+        const newHasUnsavedChangesMap = {
             ...ensembleIdentStringHasUnsavedChangesMap,
             [ensembleIdentString]: hasUnsavedChanges,
-        });
+        };
+        setEnsembleIdentStringHasUnsavedChangesMap(newHasUnsavedChangesMap);
+
+        // Update number of unsaved filters
+        const numberOfUnsavedFilters = Object.values(newHasUnsavedChangesMap).filter(
+            (hasUnsavedChanges) => hasUnsavedChanges
+        ).length;
+        setNumberOfUnsavedRealizationFilters(numberOfUnsavedFilters);
```

Comment:
You've repeated these lines a bunch of times. Could be nice to extract it into a utility function for it